File: 백준/Bronze/15818. 오버플로우와 모듈러/오버플로우와 모듈러.cc
#include <iostream>
#include <vector>
using namespace std;
int main(void) {
    int n,m;
    cin>>n>>m;
    long long rlt=1;
    vector<int> arr(n);
    for(int i=0;i<n;i++) {
        cin>>arr[i];
        rlt*=arr[i];
        rlt%=m;
    }
    cout<<rlt;
    return 0;
}

File: 백준/Bronze/15818. 오버플로우와 모듈러/오버플로우와 모듈러.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.util.Arrays;
public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        long rlt=1;
        int[] arr = Arrays.stream(br.readLine().split(" "))
        .mapToInt(Integer::parseInt).toArray();
        for(int i=0;i<n;i++) {
            rlt*=arr[i];
            rlt%=m;
        }
        System.out.print(rlt);
    }
}